lib/Flow.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Amp;
+
+/**
+ * Defines a flow, an asynchronous generator that yields key/value pairs as data is available.
+ */
+interface Flow
+{
+    /**
+     * Succeeds with a [value, key] pair or null if no more values are available. If the flow fails, the returned promise
+     * will fail with the same exception.
+     *
+     * @return \Amp\Promise<[mixed $value, mixed $key]|null>
+     *
+     * @throws \Error If the prior promise returned from this method has not resolved.
```

Comment:
I think we should allow concurrent `continue` operations in `Flow`, as that's another issue we currently have with `Iterator`.